diem-node/src/lib.rs

```diff
@@ -238,42 +250,137 @@ fn setup_debug_interface(config: &NodeConfig, logger: Option<Arc<Logger>>) -> No
+        db_rw,
+        chunk_executor,
+        node_config,
+        waypoint,
+        event_subscription_service,
+        diem_data_client,
+        streaming_service_client,
+    );
+
+    // Create and return the new state sync handle
+    StateSyncRuntimes::new(
+        diem_data_client_runtime,
+        state_sync_multiplexer,
+        storage_service_runtime,
+        streaming_service_runtime,
+    )
+}
+
+fn setup_data_streaming_service(
+    config: DataStreamingServiceConfig,
+    diem_data_client: DiemNetDataClient,
+) -> (StreamingServiceClient, Runtime) {
+    // Create the data streaming service
+    let (streaming_service_client, streaming_service_listener) =
+        new_streaming_service_client_listener_pair();
+    let data_streaming_service =
+        DataStreamingService::new(config, diem_data_client, streaming_service_listener);
+
+    // Start the data streaming service
+    let streaming_service_runtime = Builder::new_multi_thread()
+        .thread_name("data-streaming-service")
         .enable_all()
         .build()
-        .expect("Failed to start the DiemNet storage-service runtime.");
-    let storage_reader = StorageReader::new(Arc::clone(&db_rw.reader));
-    for events in network_handles {
-        let service =
-            StorageServiceServer::new(config, rt.handle().clone(), storage_reader.clone(), events);
-        rt.spawn(service.start());
-    }
-    rt
+        .expect("Failed to create data streaming service!");
+    streaming_service_runtime.spawn(data_streaming_service.start_service());
+
+    (streaming_service_client, streaming_service_runtime)
 }
 
-fn setup_diemnet_data_client(
-    _runtime_handle: &Handle,
+fn setup_diem_data_client(
     config: StorageServiceConfig,
     network_handles: HashMap<NetworkId, storage_service_client::StorageServiceNetworkSender>,
     peer_metadata_storage: Arc<PeerMetadataStorage>,
-) -> DiemNetDataClient {
+) -> (DiemNetDataClient, Runtime) {
     // Combine all storage service client handles
     let network_client = StorageServiceClient::new(
         StorageServiceMultiSender::new(network_handles),
         peer_metadata_storage,
     );
-    let (diemnet_data_client, _data_summary_poller) =
+
+    // Create the diem data client
+    let (diem_data_client, data_summary_poller) =
         DiemNetDataClient::new(config, TimeService::real(), network_client);
-    // TODO(philiphayes): uncomment this when we're ready to start doing e2e tests
-    // runtime_handle.spawn(data_summary_poller);
-    diemnet_data_client
+
+    // Create a new runtime for the diem data client and spawn the data poller
+    let diem_data_client_runtime = Builder::new_multi_thread()
+        .thread_name("diem-data-client")
+        .enable_all()
+        .build()
+        .expect("Failed to create diem data client!");
+    diem_data_client_runtime.spawn(data_summary_poller.start());
+
+    (diem_data_client, diem_data_client_runtime)
+}
+
+fn setup_storage_service(
```

Comment:
do we rip off the existing storage service? having two `storage service` for different purpose seems confusing.

Reply:
Yeah, it's not ideal. This storage service is over diem networks (between nodes), but (from what I understand), the other storage service is required by backup and restore ('cc @msmouse, is that right?) I don't know if we can unify them now, but I agree that it would be cool if we could in the future. But, I'm not sure how that work yet 🤔

Reply:
we should at least name them differently

Reply:
Name it S4 (StateSyncStorageService)

Reply:
@zekun000, rename the entire struct? Or just the methods that create the storage service for the `StateSyncRuntimes`? (Note: these storage services live in different crates/directories in the codebase so they do have some context).

I've renamed the helper methods to make it obvious that it's for state sync. If you'd prefer a rename of the entire struct, I can do that, but it'll need to happen in a new/separate PR 😄

Reply:
maybe update public names that appear outside of the state sync crate, definitely not in this pr